src/java/htsjdk/variant/variantcontext/CommonInfo.java

```diff
@@ -247,41 +284,122 @@ public Object getAttribute(String key, Object defaultValue) {
         as a java.util.List if the value is a List or an Array,
         as a Collections.singletonList if there is only one value */
     @SuppressWarnings("unchecked")
-    public List<Object> getAttributeAsList(String key) {
-        Object o = getAttribute(key);
+    public List<Object> getAttributeAsList(final String key) {
+        final Object o = getAttribute(key);
         if ( o == null ) return Collections.emptyList();
-        if ( o instanceof List ) return (List<Object>)o;
-        if ( o.getClass().isArray() ) return Arrays.asList((Object[])o);
+        else if ( o instanceof List ) return (List<Object>)o;
```

Comment:
I appreciate that this is a problem, but I hate to add more instanceof calls.  I guess that's an implicit vote for the proposal by @tfenne in #520